modules/openapi-generator/src/main/resources/typescript-redux-query/apis.mustache

```diff
@@ -0,0 +1,248 @@
+// tslint:disable
+{{>licenseInfo}}
+
+import { HttpMethods, QueryConfig, ResponseBody, ResponseText } from 'redux-query';
+import * as runtime from '../runtime';
+
+{{#imports.0}}
+import {
+    {{#imports}}
+    {{className}},
+    {{className}}FromJSON,
+    {{className}}ToJSON,
+    {{/imports}}
+} from '../models';
+{{/imports.0}}
+
+{{#operations}}
+{{#operation}}
+{{#allParams.0}}
+export interface {{operationIdCamelCase}}Request {
+    {{#allParams}}
+    {{paramName}}{{^required}}?{{/required}}: {{#isEnum}}{{{datatypeWithEnum}}}{{/isEnum}}{{^isEnum}}{{{dataType}}}{{/isEnum}};
+    {{/allParams}}
+}
+
+{{/allParams.0}}
+{{/operation}}
+{{/operations}}
+{{#operations}}
+
+{{#operation}}
+/**
+ {{#notes}}
+ * {{&notes}}
+ {{/notes}}
+ {{#summary}}
+ * {{&summary}}
+ {{/summary}}
+ */
+function {{nickname}}Raw<T>({{#allParams.0}}requestParameters: {{operationIdCamelCase}}Request,{{/allParams.0}} requestConfig: runtime.TypedQueryConfig<T, {{#returnType}}{{{returnType}}}{{/returnType}}{{^returnType}}void{{/returnType}}> = {}): QueryConfig<T> {
+    {{#allParams}}
+    {{#required}}
+    if (requestParameters.{{paramName}} === null || requestParameters.{{paramName}} === undefined) {
+        throw new runtime.RequiredError('{{paramName}}','Required parameter requestParameters.{{paramName}} was null or undefined when calling {{nickname}}.');
+    }
+
+    {{/required}}
+    {{/allParams}}
+    let queryParameters = null;
+
+    {{#queryParams}}
+    queryParameters = {};
+
+    {{#isListContainer}}
+    if (requestParameters.{{paramName}}) {
+        {{#isCollectionFormatMulti}}
+        queryParameters['{{baseName}}'] = requestParameters.{{paramName}};
+        {{/isCollectionFormatMulti}}
+        {{^isCollectionFormatMulti}}
+        queryParameters['{{baseName}}'] = requestParameters.{{paramName}}.join(runtime.COLLECTION_FORMATS["{{collectionFormat}}"]);
+        {{/isCollectionFormatMulti}}
+    }
+
+    {{/isListContainer}}
+    {{^isListContainer}}
+    if (requestParameters.{{paramName}} !== undefined) {
+        {{#isDateTime}}
+        queryParameters['{{baseName}}'] = (requestParameters.{{paramName}} as any).toISOString();
```

Comment:
add null-check

Reply:
I'm not sure I'm following this request...

1. `queryParameters` gets initialized on line 52 if we're inside this `{{#queryParameters}}` block.
1. The call to `toISOString()` only occurs inside the `{{#isDateTime}}` check.

Is this a risk for "non-required DateTime parameters" that at the TS level get typed as `foo?: Date` and about case #2 here?

Reply:
`requestParameters.{{paramName}}` could be null or undefined, so it would be better to have
```typescript
if (requestParameters.{{paramName}} !== undefined && requestParameters.{{paramName}} !== null) {
```